flytekit/common/component_nodes.py

```diff
@@ -0,0 +1,140 @@
+from __future__ import absolute_import
+
+import six as _six
+import logging as _logging
+
+from flytekit.common import sdk_bases as _sdk_bases
+from flytekit.common.exceptions import system as _system_exceptions
+from flytekit.models.core import workflow as _workflow_model
+
+
+class SdkTaskNode(_six.with_metaclass(_sdk_bases.ExtendedSdkType, _workflow_model.TaskNode)):
+
+    def __init__(self, sdk_task):
+        """
+        :param flytekit.common.tasks.task.SdkTask sdk_task:
+        """
+        self._sdk_task = sdk_task
+        super(SdkTaskNode, self).__init__(None)
+
+    @property
+    def reference_id(self):
+        """
+        A globally unique identifier for the task.
+        :rtype: flytekit.models.core.identifier.Identifier
+        """
+        return self._sdk_task.id
+
+    @property
+    def sdk_task(self):
+        """
+        :rtype: flytekit.common.tasks.task.SdkTask
+        """
+        return self._sdk_task
+
+    @classmethod
+    def promote_from_model(cls, base_model, tasks):
+        """
+        Takes the idl wrapper for a TaskNode and returns the hydrated Flytekit object for it by fetching it from the
+        engine.
+
+        :param flytekit.models.core.workflow.TaskNode base_model:
+        :param list[flytekit.models.task.TaskTemplate] tasks:
+        :rtype: SdkTaskNode
+        """
+        from flytekit.common.tasks import task as _task
+        tasks = tasks or []
```

Comment:
nit: if we're creating this before passing into the function, maybe we make it a dictionary of id -> task in the caller and use it for all tasks.